src/app/services/ServerApi.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class ServerApiService {
+  // TODO: Configure base Url, to point to hosted endpoint. (localhost:3333 isn't a valid endpoint.)
+  private baseUrl = 'https://jsonplaceholder.typicode.com/todos/1';  // URL to web api
+
+  constructor(private http: HttpClient) { }
+
+  // Base Function
+  private getFromDataBase(urlExtension: string): Observable<any> {
```

Comment:
Presumably this should be public?

Reply:
This could be public (However, this would allow Angular Components to access the base function.) One of the suggested practices I observed in the angular documentation was to add other public functions that handle the specific GET/POST calls. (This is what getTests is referring to in the ClientApiInstructions.)